crates/autopilot/src/domain/settlement/auction.rs

```diff
@@ -0,0 +1,27 @@
+//! Auction data related to the specific settlement.
+
+use {
+    crate::domain::{self, competition, eth},
+    std::collections::{HashMap, HashSet},
+};
+
+/// Offchain auction data related to a specific settlement that got settled for
+/// it.
+#[derive(Debug)]
+pub struct Auction {
+    /// Competition winner (solver submission address).
+    pub winner: eth::Address,
+    /// Settlement should appear onchain before this block.
+    pub deadline: eth::BlockNo,
+    /// Winning score promised during competition (based on the promised
+    /// `competition::Solution`)
+    pub score: competition::Score,
+    /// Winning solution promised during competition.
+    pub solution: competition::Solution,
+    /// Auction external prices
+    pub prices: domain::auction::Prices,
+    /// Settlement orders that are missing from the orderbook (JIT orders).
+    pub missing_orders: HashSet<domain::OrderUid>,
```

Comment:
Should this be allowed to include orders that are in the orderbook but were not part of the original auction? IMO these would technically not be JIT orders.
Coming to think of it if a solver settles a user order that was not part of the auction it should owe the protocol the order fees on top of whatever amount it gets slashed for. 🤔

Reply:
[Should this be allowed to include orders that are in the orderbook but were not part of the original auction?](url)

IMO no. 
1. A promised solution (solution returned during competition) that contains out-of-auction orders should be rejected. 
2. If a promised solution doesn't contain any out-of-auction order, but delivered solution does, solver should be removed.